hs_geographic_feature_resource/models.py

```diff
@@ -112,26 +289,31 @@ class Meta:
 processor_for(GeographicFeatureResource)(resource_processor)
 
 
-# define the GeographicFeatureMetaData metadata
-class GeographicFeatureMetaData(CoreMetaData):
-    geometryinformation = GenericRelation(GeometryInformation)
-    fieldinformation = GenericRelation(FieldInformation)
-    originalcoverage = GenericRelation(OriginalCoverage)
-    originalfileinfo = GenericRelation(OriginalFileInfo)
+class GeographicFeatureMetaDataMixin(models.Model):
+    """This class must be the first class in the multi-inheritance list of classes"""
+    geometryinformations = GenericRelation(GeometryInformation)
+    fieldinformations = GenericRelation(FieldInformation)
+    originalcoverages = GenericRelation(OriginalCoverage)
+
+    class Meta:
+        abstract = True
 
     @property
-    def resource(self):
-        return GeographicFeatureResource.objects.filter(object_id=self.id).first()
+    def geometryinformation(self):
+        return self.geometryinformations.all().first()
+
+    @property
+    def originalcoverage(self):
+        return self.originalcoverages.all().first()
 
     @classmethod
     def get_supported_element_names(cls):
         # get the names of all core metadata elements
-        elements = super(GeographicFeatureMetaData, cls).get_supported_element_names()
+        elements = super(GeographicFeatureMetaDataMixin, cls).get_supported_element_names()
```

Comment:
@pkdash This is confusing to me. It seems to me then if these methods such as get_supported_element_names() are already in ```CoreMetaData``` class, then there is no need to add these methods in the Mixin class. I might have missed something, but I don't get why these methods are needed in the Mixin class.